src/test/java/io/stargate/sgv2/jsonapi/api/v1/CountIntegrationTest.java

```diff
@@ -0,0 +1,675 @@
+package io.stargate.sgv2.jsonapi.api.v1;
+
+import static io.restassured.RestAssured.given;
+import static io.stargate.sgv2.common.IntegrationTestUtils.getAuthToken;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+import io.restassured.http.ContentType;
+import io.stargate.sgv2.api.common.config.constants.HttpConstants;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@QuarkusIntegrationTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class CountIntegrationTest extends CollectionResourceBaseIntegrationTest {
+  @Nested
+  @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+  class Count {
+    @Test
+    @Order(1)
+    public void setUp() {
+      String json =
+          """
+                                          {
+                                            "insertOne": {
```

Comment:
I think a helper method for inserting a single document would make sense; just take document contents, verify call succeeds. These tests are getting quite verbose otherwise.

Reply:
Make sense, changed the code.